bin/ubuntu-core-initramfs

```diff
@@ -616,6 +611,58 @@ def create_initrd_pkg_list(dest_dir, sysroot):
                             pkgs).decode("utf-8")
         pkg_list.write(out)
 
+def verify_missing_dlopen(destdir, libdir):
+    missing = {}
+    for dirpath, dirs, files in os.walk(destdir):
+        for f in files:
+            path = os.path.join(dirpath, f)
+            if os.path.islink(path) or not os.path.isfile(path):
+                continue
+            with open(path, 'rb') as b:
+                if b.read(4) != b'\x7fELF':
+                    continue
+            out = check_output(["dlopen-notes", path])
+            split = out.splitlines()
+            json_doc = b'\n'.join([s for s in split if not s[:1] == b'#'])
+            doc = json.loads(json_doc)
+            for dep in doc:
+                sonames = dep["soname"]
+                priority = dep["priority"]
+                found_sonames = []
+                for soname in sonames:
+                    dest = os.path.join(destdir, os.path.relpath(libdir, "/"), soname)
+                    if os.path.exists(os.path.join(destdir, dest)):
+                        found_sonames.append(soname)
+                if not found_sonames:
+                    for soname in sonames:
+                        current_priority = missing.get(soname)
+                        if current_priority == "required":
+                            continue
+                        elif current_priority == "recommended" and priority not in ["required"]:
+                            continue
+                        elif current_priority == "suggested" and priority not in ["required", "recommended"]:
+                            continue
+                        else:
+                            missing[soname] = priority
+
+    fatal = False
+    if missing:
+        print(f"WARNING: These sonames are missing:", file=sys.stderr)
+        for m, priority in missing.items():
+            print(f" * {m} ({priority})", file=sys.stderr)
+            if priority == "required":
```

Comment:
I wonder if we should require any "recommended" library. I see there libkmod.so.2 and libzstd.so.1 and both are probably useful according to "feature".

Reply:
I think at this point only "required" is used. But maybe I have missed something.

Reply:
I see some recommended in oracle:
```
root@oracular-confined:~# dlopen-notes -s /usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-256.so  | grep recommended
libkmod.so.2 recommended
libzstd.so.1 recommended
```

Reply:
I was wrong, "required" is never used by systemd.  I will make missing "recommended" libraries fatal.